File: ocaml/difference-of-squares/difference_of_squares.ml
let rec range acc n =
    match n with
    0 -> acc |
    _ -> range (n::acc) (n - 1)

let sq n = n * n

let sum = List.fold_left (+) 0

let square_of_sum n =
    range [] n
    |> sum
    |> sq

let sum_of_squares n =
    range [] n
    |> List.map sq
    |> sum

let difference_of_squares n =
    square_of_sum n - sum_of_squares n

File: lua/difference-of-squares/difference-of-squares.lua
local function square_of_sum(n)
    local s = 0
    for i = 1, n do
        s = s + i
    end

    return s ^ 2

end

local function sum_of_squares(n)

    local s = 0
    for i = 1, n do
        s = s + i ^ 2
    end

    return s

end

local function difference_of_squares(n)

    return square_of_sum(n) - sum_of_squares(n)

end

return {
    square_of_sum = square_of_sum,
    sum_of_squares = sum_of_squares,
    difference_of_squares = difference_of_squares
}
